build-packages/scandipwa-template-plugin-system/errors/both-api-usage.js

```diff
@@ -0,0 +1,14 @@
+const logger = require('@scandipwa/scandipwa-dev-utils/logger');
+const { DOM_API_KEY, STRING_API_KEY } = require('../util/exposed-api-keys');
+
+module.exports = class NoPluginReturnDomException extends Error {
+    constructor() {
+        super();
+
+        this.message = `
```

Comment:
So you created all these types of error, thus you are throwing them only once each and catching them all together.

I am not sure the error class extension makes sense as not much functionality lies into this logic. You can instead throw simple Errors. I think creating error clasess is usefull if you plan to handle these exceptions differently.

I think it can stay, but I would say it does look like "overingeneered" solution.